dotCMS/src/main/java/com/dotmarketing/util/ImportExportUtil.java

```diff
@@ -1140,113 +1151,127 @@ private void doXMLFileImport(File f, PrintWriter out, ObjectFilter filter)throws
-                    dc.getResults();
+                    final HashMap<String, String> dcResults = (HashMap<String,String>)l.get(j);
+                    LocalTransaction.wrap(() -> {
+                        DotConnect dc = new DotConnect();
+                        dc.setSQL("insert into pollsdisplay values (?,?,?,?)");
+                        dc.addParam(dcResults.get("layoutid"));
+                        dc.addParam(dcResults.get("userid"));
+                        dc.addParam(dcResults.get("portletid"));
+                        dc.addParam(dcResults.get("questionid"));
+                        dc.getResults();
+                    });
                 }
             }else if(pollsquestion){
                 for (int j = 0; j < l.size(); j++) {
-                    HashMap<String, String> dcResults = (HashMap<String,String>)l.get(j);
-                    DotConnect dc = new DotConnect();
-                    dc.setSQL("insert into pollsquestion values (?,?,?,?,?,?,?,?,?,?,?,?)");
-                    dc.addParam(dcResults.get("questionid"));
-                    dc.addParam(dcResults.get("portletid"));
-                    if(UtilMethods.isSet(dcResults.get("groupid"))){
-                        dc.addParam(dcResults.get("groupid"));
-                    }else{
-                        dc.addParam(-1);
-                    }
-                    dc.addParam(dcResults.get("companyid"));
-                    dc.addParam(dcResults.get("userid"));
-                    dc.addParam(dcResults.get("username"));
-                    if(UtilMethods.isSet(dcResults.get("createdate"))){
-                        dc.addParam(java.sql.Timestamp.valueOf(dcResults.get("createdate")));
-                    }else{
-                        dc.addParam(new java.sql.Timestamp(0));
-                    }
-                    if(UtilMethods.isSet(dcResults.get("modifieddate"))){
-                        dc.addParam(java.sql.Timestamp.valueOf(dcResults.get("modifieddate")));
-                    }else{
-                        dc.addParam(new java.sql.Timestamp(0));
-                    }
-                    dc.addParam(dcResults.get("title"));
-                    dc.addParam(dcResults.get("description"));
-                    if(UtilMethods.isSet(dcResults.get("expirationdate"))){
-                        dc.addParam(java.sql.Timestamp.valueOf(dcResults.get("expirationdate")));
-                    }else{
-                        dc.addParam(new java.sql.Timestamp(0));
-                    }
-                    if(UtilMethods.isSet(dcResults.get("lastvotedate"))){
-                        dc.addParam(java.sql.Timestamp.valueOf(dcResults.get("lastvotedate")));
-                    }else{
-                        dc.addParam(new java.sql.Timestamp(0));
-                    }
-                    dc.getResults();
+                    final HashMap<String, String> dcResults = (HashMap<String,String>)l.get(j);
+                    LocalTransaction.wrap(() -> {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this lambda expression number of lines from 37 to at most 20. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1188)